src/main/java/org/websoso/WSSServer/service/UserNovelService.java

```diff
@@ -332,4 +336,51 @@ private List<Genre> getPriorityOrderByGender(Gender gender, List<Genre> allGenre
                 )
                 .toList();
     }
+
+    @Transactional(readOnly = true)
+    public UserTasteAttractivePointPreferencesAndKeywordsGetResponse getUserAttractivePointsAndKeywords(User visitor,
+                                                                                                        Long ownerId) {
+        User owner = userService.getUserOrException(ownerId);
+
+        if (owner.getIsProfilePublic() || isOwner(visitor, ownerId)) {
+            List<UserNovel> ownerUserNovels = userNovelRepository.findUserNovelByUser(owner);
+
+            Map<String, Long> ownerAttractivePointCountMap = ownerUserNovels.stream()
+                    .map(UserNovel::getUserNovelAttractivePoints)
+                    .flatMap(List::stream)
+                    .map(UserNovelAttractivePoint::getAttractivePoint)
+                    .collect(Collectors.groupingBy(AttractivePoint::getAttractivePointName, Collectors.counting()));
+
+            List<String> top3OwnerAttractivePointNames = ownerAttractivePointCountMap.entrySet()
+                    .stream()
+                    .sorted(Entry.<String, Long>comparingByValue().reversed())
+                    .limit(3)
+                    .map(Entry::getKey)
+                    .toList();
```

Comment:
p2; 아래처럼 한번에 구할 수도 있지 않을까요? 단계로 나눈 이유가 있으신가요? 그리고 이렇게 getTop3AttractivePoints()와 같이 메서드를 분리하면 좋을 것 같습니다
```suggestion
private List<String> getTop3AttractivePoints(List<UserNovel> userNovels) {
    return userNovels.stream()
            .map(UserNovel::getUserNovelAttractivePoints)
            .flatMap(List::stream)
            .map(UserNovelAttractivePoint::getAttractivePoint)
            .collect(Collectors.groupingBy(AttractivePoint::getAttractivePointName, Collectors.counting()))
            .entrySet()
            .stream()
            .sorted(Map.Entry.<String, Long>comparingByValue().reversed())
            .limit(3)
            .map(Map.Entry::getKey)
            .collect(Collectors.toList());
}
```

Reply:
음 무조건 한 번에 처리하는게 좋다고 생각하지는 않아서요, 논리적 단위에 맞게 구분해서 처리했습니다.
재사용되는 곳이 추가로 있는 곳이 있다면 메서드 추출도 논리적 단위에 맞게 하는게 낫다고 생각하구욥.

메서드 추출도 재사용성이나 유지보수 측면에서 이점이 있기에 사용하면 좋은거지 꼭 해야할 필요가 있을까욥?

---

1. 스트림 같은 체이닝을 한 번에 처리하는 것이 좋은지? 그렇다면 그 이유는 무엇인지?
2. 메서드 추출은 항상 좋은지? 그렇다면 이유는 무엇인지?
에 대한 생각이 궁금하네요~~

Reply:
1> 한번에 처리하는게 더 간결해보인다고 생각했는데 이건 사람마다 다를 수 있겠네요
2> 매력포인트와 키워드 모두 추출하는 과정이 길어서 해당 상황에서는 메서드를 추출하는게 더 깔끔할 것 같다고 생각했습니다. 근데 이것도 마찬가지로 꼭 필요하지 않다고 생각하신다면..? 굳이긴 하네요
코멘트 확인하시면 resolve 해주세욥!!

Reply:
나중에 재사용하게 된다면 메서드로 추출하겠씁니닷!
(재사용을 고려해 이미 생성된 private 메서드들이 잘 사용되지 않는 것 같기도 해서 -> 한 번 전체적인 리팩터링이 필요할 것 같습니닷.)